bundles/org.openhab.binding.mikrotik/src/main/java/org/openhab/binding/mikrotik/internal/handler/MikrotikRouterosBridgeHandler.java

```diff
@@ -0,0 +1,268 @@
+            // Perform connection
+            try {
+                logger.debug("Starting routeros model");
+                routeros.start();
+
+                @Nullable
+                RouterosRouterboardInfo rbInfo = routeros.getRouterboardInfo();
+                if (rbInfo != null) {
+                    Map<String, String> bridgeProps = editProperties();
+                    bridgeProps.put(MikrotikBindingConstants.PROPERTY_MODEL, rbInfo.getModel());
+                    bridgeProps.put(MikrotikBindingConstants.PROPERTY_FIRMWARE, rbInfo.getFirmware());
+                    bridgeProps.put(MikrotikBindingConstants.PROPERTY_SERIAL_NUMBER, rbInfo.getSerialNumber());
+                    updateProperties(bridgeProps);
+                } else {
+                    logger.warn("Failed to set RouterBOARD properties for bridge {}", getThing().getUID());
+                }
+                updateStatus(ThingStatus.ONLINE);
+            } catch (MikrotikApiException e) {
+                logger.warn("Error while logging in to RouterOS {}", getThing().getUID(), e);
+                logger.debug("RouterOS connection exception cause", e.getCause());
+
+                if (e.getMessage().contains("Command timed out") || e.getMessage().contains("Error connecting")) {
+                    routeros.stop();
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+                } else if (e.getMessage().contains("Connection refused")) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                            "Remote host refused to connect, make sure port is correct");
+                } else {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
+                }
+            }
+        } else {
+            // We're connected - do a usual polling cycle
+            performRefresh();
+        }
+    }
+
+    private void performRefresh() {
+        try {
+            logger.debug("Refreshing RouterOS caches for {}", getThing().getUID());
+            routeros.refresh();
+            // refresh own channels
+            for (Channel channel : getThing().getChannels()) {
+                try {
+                    refreshChannel(channel.getUID());
+                } catch (RuntimeException e) {
+                    throw new ChannelUpdateException(getThing().getUID(), channel.getUID(), e);
+                }
+            }
+            // refresh all the client things below
+            getThing().getThings().forEach(thing -> {
+                ThingHandler handler = thing.getHandler();
+                if (handler instanceof MikrotikBaseThingHandler) {
+                    ((MikrotikBaseThingHandler) handler).refresh();
+                }
+            });
+        } catch (ChannelUpdateException e) {
+            logger.error("Error updating channel! {}", e.getMessage(), e.getCause());
```

Comment:
Bindings should only log to error if something severe happened, like the detection of a bug in your code. This could be debug. See this link for a description of the log levels: https://www.openhab.org/docs/developer/guidelines.html#f-logging

Please check all.